eng/common/instructions/azsdk-tools/local-sdk-workflow.instructions.md

```diff
@@ -0,0 +1,119 @@
+---
+description: "Guide the user to generate and build SDKs locally for TypeSpec changes"
+---
+
+# Goal
+
+Help the user generate and build SDKs locally from TypeSpec API specifications using the `azure-sdk-mcp` tools.
+
+---
+
+## Part A: Generate SDK Locally
+
+### Step 1: Outline workflow
+
+**Goal**: Ensure the user understands the overall SDK generation and build process before starting.  
+**Actions**:
+
+- Present the high-level steps involved in generating and building SDK locally:
+  1. Select target language
+  2. Verify SDK repository
+  3. Validate repository path
+  4. Identify path to configuration file
+  5. Generate SDK using `azure-sdk-mcp`
+  6. Identify SDK project path
+  7. Build/Compile SDK using `azure-sdk-mcp`
+- Ask the user to confirm readiness to proceed.
+
+---
+
+### Step 2: Select language
+
+**Goal**: Confirm the target language for SDK generation.  
+**Actions**:
+
+- Prompt user to choose one of the supported languages:
+  - .NET
+  - Java
+  - JavaScript
+  - Python
+  - Go
+- Validate input against the allowed list.
+
+---
+
+### Step 3: Verify SDK repository
+
+**Goal**: Ensure the correct Azure SDK language repository is available locally.
+**Actions**:
+
+- Prompt user to provide the path to their **locally cloned repository** for the selected language.
+- Note: The **local folder name can be arbitrary**, but the repository must have originated from one of the official Azure SDK repositories:
+  - `azure-sdk-for-net` (.NET)
+  - `azure-sdk-for-java` (Java)
+  - `azure-sdk-for-js` (JavaScript)
+  - `azure-sdk-for-python` (Python)
+  - `azure-sdk-for-go` (Go)
+- If the repository is not cloned → instruct user to clone the appropriate remote repository from GitHub.
+- MCP tool will automatically validate the remote origin and repository structure.
+
+---
+
+### Step 4: Validate repository path
+
+**Actions**:
+
+- Check if the provided repository path exists and matches the selected SDK language repository.
```

Comment:
Do we need to verify the setup in language repo? I remember @LibbaLawrence had a similar step in old version of local SDK generation for Python

Reply:
Setup verification is not covered here. When we have the `setup-env` tool ready, it will be chained to these two tools if the error is related to env issue.